syntax = "proto3";

// The service definition.
service UI {
    rpc Ping (Empty) returns (SuccessReply) {};
    rpc ImageUpdate (ImageUpdateRequest) returns (SuccessReply);
    rpc Exit (Empty) returns (Empty);
}
  
message Empty {
}

message SuccessReply {
    bool ok = 1;
}

message PingReply {
    bool ok = 1;
}

message ImageUpdateRequest {
    int32 image_index = 1;
    int32 mask_index = 2;
    int32 timestamp = 3;
    string camera_id = 4;
    string capture_config = 5;
    int32 capture_index = 6;
    int32 infer_timestamp = 7;
    string network = 8;
    bool ok = 9;
    string part_id = 10;
}